scipy/stats/_hypotests_pythran.py

```diff
@@ -149,3 +150,117 @@ def _compute_outer_prob_inside_method(m, n, g, h):
+        0 <= h <= n
+
+    Returns
+    -------
+    p : float
+        The proportion of paths that pass outside the lines x-y = +/-h.
+
+    """
+    # Compute Pr(D_{n,n} >= h/n)
+    # Prob = 2 * ( binom(2n, n-h) - binom(2n, n-2a) + binom(2n, n-3a) - ... )
+    # / binom(2n, n)
+    # This formulation exhibits subtractive cancellation.
+    # Instead divide each term by binom(2n, n), then factor common terms
+    # and use a Horner-like algorithm
+    # P = 2 * A0 * (1 - A1*(1 - A2*(1 - A3*(1 - A4*(...)))))
+
+    P = 0.0
+    k = int(np.floor(n / h))
+    while k >= 0:
+        p1 = 1.0
+        # Each of the Ai terms has numerator and denominator with
+        # h simple terms.
+        for j in range(h):
+            p1 = (n - k * h - j) * p1 / (n + k * h + j + 1)
+        P = p1 * (1.0 - P)
+        k -= 1
+    return 2 * P
+
+
+# pythran export _count_paths_outside_method(int64, int64, int64, int64)
+def _count_paths_outside_method(m, n, g, h):
+    """Count the number of paths that pass outside the specified diagonal.
+
+    Parameters
+    ----------
+    m : integer
+        m > 0
+    n : integer
+        n > 0
+    g : integer
+        g is greatest common divisor of m and n
+    h : integer
+        0 <= h <= lcm(m,n)
+
+    Returns
+    -------
+    p : float
+        The number of paths that go low.
+        The calculation may overflow - check for a finite answer.
+
+    Notes
+    -----
+    Count the integer lattice paths from (0, 0) to (m, n), which at some
+    point (x, y) along the path, satisfy:
+      m*y <= n*x - h*g
+    The paths make steps of size +1 in either positive x or positive y
+    directions.
+
+    We generally follow Hodges' treatment of Drion/Gnedenko/Korolyuk.
+    Hodges, J.L. Jr.,
+    "The Significance Probability of the Smirnov Two-Sample Test,"
+    Arkiv fiur Matematik, 3, No. 43 (1958), 469-86.
+
+    """
+    # Compute #paths which stay lower than x/m-y/n = h/lcm(m,n)
+    # B(x, y) = #{paths from (0,0) to (x,y) without
+    #             previously crossing the boundary}
+    #         = binom(x, y) - #{paths which already reached the boundary}
+    # Multiply by the number of path extensions going from (x, y) to (m, n)
+    # Sum.
+
+    # Probability is symmetrical in m, n.  Computation below assumes m >= n.
+    if m < n:
+        m, n = n, m
+    mg = m // g
+    ng = n // g
+
+    # Not every x needs to be considered.
+    # xj holds the list of x values to be checked.
+    # Wherever n*x/m + ng*h crosses an integer
+    lxj = n + (mg-h)//mg
+    xj = [(h + mg * j + ng-1)//ng for j in range(lxj)]
+    # B is an array just holding a few values of B(x,y), the ones needed.
+    # B[j] == B(x_j, j)
+    if lxj == 0:
+        return binom(m + n, n)
+    B = np.zeros(lxj)
+    B[0] = 1
+    # Compute the B(x, y) terms
+    for j in range(1, lxj):
+        Bj = binom(xj[j] + j, j)
+        for i in range(j):
+            bin = binom(xj[j] - xj[i] + j - i, j-i)
+            Bj -= bin * B[i]
```

Comment:
> Oh, do you mean that the invalid call to binom should raise an Exception instead of returning nan ?

No, I wasn't requesting any changes to the behavior of `binom`. I'm not necessarily saying that Pythran is doing anything wrong, either. It's just different from what pure Python/NumPy does, and I'm asking for your suggestion about what I should do.

Let me describe what happens when this code runs in pure Python/NumPy rather than via Pythran.

This line - 258 - is where the code errors out. In the example code, at `j=367` and `i=366`, both `Bj` and `B[i]` have value `np.float64(np.inf)`, and there is a subtraction that would result in a `nan`. Outside of the `np.errstate` context, for example, something like:
```python3
x = np.float64(np.inf)
x - x
```
results in
```
C:\Users\matth\AppData\Local\Temp\ipykernel_20332\4213061242.py:1: RuntimeWarning: invalid value encountered in double_scalars
  x-x
nan
```
But within the `np.errstate` context that this function is in, this line actually raises:
```
FloatingPointError: invalid value encountered in double_scalars
```
(Previously, I said that there was an _overflow_ error. Sorry if that was misleading; that is not exactly what is happening.)

When this code runs via Pythran, I don't get the same `FloatingPointError`, but instead I get the "RuntimeError: Something happened on the way to Heaven". Because the code is not currently `except`ing `RuntimeError`s, the whole function errors out, so I need to change something. I was asking what you'd recommend that I change.

The obvious solution is to `except` the `RuntimeError`. But `RuntimeError` is very non-specific - I don't know in general whether the root cause is the same as the `FloatingPointError` I was getting in Python, or whether it is something else that I should know about (that perhaps can/should be fixed). So I'm hesitant to do the obvious thing here for the same sort of reason we don't often use bare `except`s: they can cover up real problems that are different from the one you're expecting.

What do you think I should do? Should I just `except` the `RuntimeError`?